### PR TITLE
Changelog for 21.8

### PR DESCRIPTION
**Pull request by filimonov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
History of edits: https://gist.github.com/johnhummelAltinity/917a7bc9183d73558ada92930c2ed941/revisions

Detailed description / Documentation draft:
It seems some edits by John @johnhummelAltinity should be used everywhere in changelog (also in future) like 'Backward Incompatible Change' -> 'Backward Incompatible Changes' etc.

To be merged when 21.8 will be ready.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,119 @@
+### ClickHouse release v21.8.2.19-prestable FIXME!! as compared to v21.7.5.29-stable
+
+#### Backward Incompatible Changes
+
+* Use Map data type for system logs tables (`system.query_log`, `system.query_thread_log`, `system.processes`, `system.OpenTelemetrySpanLog`). These tables will be auto-created with new datatypes; old queries may be incompatible. [#23934](https://github.com/ClickHouse/ClickHouse/pull/23934) ([hexiaoting](https://github.com/hexiaoting)).
```
Comment (alexey-milovidov) [inline CHANGELOG.md:5]: In fact we made it compatible.
Only the queries with SELECT * may suffer but it's alright.

Comment (vdimir) [reply CHANGELOG.md]: I merged this two entries in changelog, and removed from "Backward Incompatible Changes":  [`70db201` (#27271)](https://github.com/ClickHouse/ClickHouse/pull/27271/commits/70db20172ec91e9754e964a39eb980885d589bc4)

**Comment by filimonov** [comment]
> 'Map type for system logs tables' is not backward incompatible in 21.8 changelog

@vdimir It makes direct downgrade impossible, so it is.

**Comment by vdimir** [comment]
> @vdimir It makes direct downgrade impossible, so it is.

@filimonov does it mean that it isn't forward compatible, but not backward ?

**Comment by alex-zaitsev** [comment]
@vdimir , one can upgrade but can not downgrade without removing system tables.

**Pull request merged**